src/types/UiaTextRangeBase.cpp

```diff
@@ -310,14 +310,319 @@ IFACEMETHODIMP UiaTextRangeBase::ExpandToEnclosingUnit(_In_ TextUnit unit) noexc
     CATCH_RETURN();
 }
 
-// we don't support this currently
-IFACEMETHODIMP UiaTextRangeBase::FindAttribute(_In_ TEXTATTRIBUTEID /*textAttributeId*/,
-                                               _In_ VARIANT /*val*/,
-                                               _In_ BOOL /*searchBackward*/,
-                                               _Outptr_result_maybenull_ ITextRangeProvider** /*ppRetVal*/) noexcept
+// Method Description:
+// - Generate an attribute verification function for that attributeId and sub-type
+// Arguments:
+// - attributeId - the UIA text attribute identifier we're looking for
+// - val - the attributeId's sub-type we're looking for
+// Return Value:
+// - a function that can be used to verify if a given TextAttribute meets the attributeId's sub-type
+std::function<bool(const TextAttribute&)> UiaTextRangeBase::_getAttrVerificationFn(TEXTATTRIBUTEID attributeId, VARIANT val) const
 {
-    UiaTracing::TextRange::FindAttribute(*this);
-    return E_NOTIMPL;
+    // Most of the attributes we're looking for just require us to check TextAttribute.
+    // So if we support it, we'll return a function to verify if the TextAttribute
+    // has the desired attribute.
+    switch (attributeId)
+    {
+    case UIA_FontWeightAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The font weight can be any value from 0 to 900.
+        // The text buffer doesn't store the actual value,
+        // we just store "IsBold" and "IsFaint".
+        const auto queryFontWeight{ val.iVal };
+
+        if (queryFontWeight > FW_NORMAL)
+        {
+            // we're looking for a bold font weight
+            return [](const TextAttribute& attr) {
+                return attr.IsBold();
+            };
+        }
+        else if (queryFontWeight < FW_NORMAL)
+        {
+            // we're looking for a faint font weight
+            return [](const TextAttribute& attr) {
+                return attr.IsFaint();
+            };
+        }
+        else
+        {
+            // we're looking for "normal" font weight
+            return [](const TextAttribute& attr) {
+                return !attr.IsBold() && !attr.IsFaint();

```

Comment:
per our discussion -- remember you can have bold AND faint. wonder how that adds up for UIA.

Reply:
Removed my assumption of "faint" being a part of "font weight". Now we just check for "IsBold".